colour/io/luts/clf.py

```diff
@@ -0,0 +1,434 @@
+        apply_log_internal,
+        params,
+        extra_args,
+    )
+
+
+def mon_curve_forward(x, exponent, offset):
+    x_break = offset / (exponent - 1)
+    s = ((exponent - 1) / offset) * (
+        (offset * exponent) / ((exponent - 1) * (1 + offset))
+    ) ** exponent
+    return np.where(x >= x_break, ((x + offset) / (1 + offset)) ** exponent, x * s)
+
+
+def mon_curve_reverse(x, exponent, offset):
+    y_break = ((offset * exponent) / ((exponent - 1) * (1 + offset))) ** exponent
+    s = ((exponent - 1) / offset) * (
+        (offset * exponent) / ((exponent - 1) * (1 + offset))
+    ) ** exponent
+    return np.where(x >= y_break, (1 + offset) * x ** (1 / exponent) - offset, x / s)
+
+
+def apply_exponent_internal(
+    value: NDArrayFloat, params: clf.ExponentParams, extra_args
+) -> NDArrayFloat:
+    exponent = params.exponent
+    offset = params.offset
+    style = extra_args
+    match style:
+        case ExponentStyle.BASIC_FWD:
+            return np.maximum(0.0, value) ** exponent
+        case ExponentStyle.BASIC_REV:
+            return np.maximum(0.0, value) ** (1 / exponent)
+        case ExponentStyle.BASIC_MIRROR_FWD:
+            return np.where(
+                value >= 0,
+                value**exponent,
+                -((-value) ** exponent),  # TODO check for type in reference
+            )
+        case ExponentStyle.BASIC_MIRROR_REV:
+            return np.where(
+                value >= 0,
+                value ** (1 / exponent),
+                -((-value) ** (1 / exponent)),
+            )
+        case ExponentStyle.BASIC_PASS_THRU_FWD:
+            return np.where(value >= 0, value**exponent, value)
+        case ExponentStyle.BASIC_PASS_THRU_REV:
+            return np.where(
+                value >= 0,
+                value ** (1 / exponent),
+                value,
+            )
+        case ExponentStyle.MON_CURVE_FWD:
+            return mon_curve_forward(value, exponent, offset)
+        case ExponentStyle.MON_CURVE_REV:
+            return mon_curve_reverse(value, exponent, offset)
+        case ExponentStyle.MON_CURVE_MIRROR_FWD:
+            return np.where(
+                value >= 0,
+                mon_curve_forward(value, exponent, offset),
+                -mon_curve_forward(-value, exponent, offset),
+            )
+        case ExponentStyle.MON_CURVE_MIRROR_REV:
+            return np.where(
+                value >= 0,
+                mon_curve_reverse(value, exponent, offset),
+                -mon_curve_reverse(-value, exponent, offset),
+            )
+        case _:
+            raise ValueError(f"Invalid Exponent Style: {style}")
+
+
+def apply_exponent(node: clf.Exponent, normalised_value: NDArrayFloat) -> NDArrayFloat:
+    style = node.style
+    params = node.exponent_params
+    return apply_by_channel(
+        normalised_value, apply_exponent_internal, params, extra_args=style
+    )
+
+
+def asc_cdl_luma(value):
+    R, G, B = tsplit(value)
```

Comment:
**Minor**

We should be able to do something along the line of `np.sum(weights * value, axis=-1)` to avoid the `tsplit`.